lib/jekyll/regenerator.rb

```diff
@@ -104,7 +104,10 @@ def modified?(path)
     def add_dependency(path, dependency)
       return if (metadata[path].nil? || @disabled)
 
-      metadata[path]["deps"] << dependency unless metadata[path]["deps"].include? dependency
+      if !metadata[path]["deps"].include? dependency
```

Comment:
maybe it should be a `Set` instead of just an `Array`? Does that slow it down unacceptably? That would eliminate the need for the check here.